samtranslator/model/eventsources/push.py

```diff
@@ -1218,38 +1218,53 @@ def _add_auth_to_openapi_integration(self, api, editor):
         # Default auth should already be applied, so apply any other auth here or scope override to default
         api_authorizers = api_auth and api_auth.get("Authorizers")
 
-        if method_authorizer != "NONE" and not api_authorizers:
-            raise InvalidEventException(
-                self.relative_id,
-                "Unable to set Authorizer [{authorizer}] on API method [{method}] for path [{path}] "
-                "because the related API does not define any Authorizers.".format(
-                    authorizer=method_authorizer, method=self.Method, path=self.Path
-                ),
-            )
+        # The IAM authorizer is built-in and not defined as a regular Authorizer.
+        enable_iam_authorizer = api_auth and api_auth.get("EnableIamAuthorizer", False) is True
 
-        if method_authorizer != "NONE" and not api_authorizers.get(method_authorizer):
-            raise InvalidEventException(
-                self.relative_id,
-                "Unable to set Authorizer [{authorizer}] on API method [{method}] for path [{path}] "
-                "because it wasn't defined in the API's Authorizers.".format(
-                    authorizer=method_authorizer, method=self.Method, path=self.Path
-                ),
-            )
+        if method_authorizer == "AWS_IAM":
+            if not enable_iam_authorizer:
```

Comment:
Why fail if IAM auth is not enabled?

Reply:
I guess we don't need to. In fact if they have a custom authorizer named "AWS_IAM" this code change might actually break them. I'll remove it.